src/parallel_progress.jl

```diff
@@ -0,0 +1,364 @@
+
+mutable struct ParallelProgress <: AbstractProgress
+    channel
+end
+
+@enum ProgressAction begin
+    PP_NEXT
+    PP_CANCEL
+    PP_FINISH
+    PP_UPDATE
+    MP_ADD_THRESH
+    MP_ADD_UNKNOWN
+    MP_ADD_PROGRESS
+end
+
+next!(pp::ParallelProgress, args...; kw...) = (put!(pp.channel, (PP_NEXT, args, kw)); nothing)
+cancel(pp::ParallelProgress, args...; kw...) = (put!(pp.channel, (PP_CANCEL, args, kw)); nothing)
+finish!(pp::ParallelProgress, args...; kw...) = (put!(pp.channel, (PP_FINISH, args, kw)); nothing)
+update!(pp::ParallelProgress, args...; kw...) = (put!(pp.channel, (PP_UPDATE, args, kw)); nothing)
+
+"""
+`ParallelProgress(n; kw...)`
+
+works like `Progress` but can be used from other workers
+
+# Example:
+```julia
+using Distributed
+addprocs()
+@everywhere using ProgressMeter
+prog = ParallelProgress(10; desc="test ")
+pmap(1:10) do
+    sleep(rand())
+    next!(prog)
+    return myid()
+end
+```
+"""
+function ParallelProgress(n::Integer; kw...)
+    return ParallelProgress(Progress(n; kw...))
+end
+
+"""
+`ParallelProgress(p::AbstractProgress)`
+
+wrapper around any `Progress`, `ProgressThresh` and `ProgressUnknown` that can be used 
+from other workers
+
+"""
+function ParallelProgress(progress::AbstractProgress)
+    channel = RemoteChannel(() -> Channel{NTuple{3,Any}}(1024))
+    pp = ParallelProgress(channel)
+    
+    @async begin
+        try
+            while !has_finished(progress) && !has_finished(pp)
+                f, args, kw = take!(channel)
+                if f == PP_NEXT
+                    next!(progress, args...; kw...)
+                elseif f == PP_CANCEL
+                    cancel(progress, args...; kw...)
+                    break
+                elseif f == PP_FINISH
+                    finish!(progress, args...; kw...)
+                    break
+                elseif f == PP_UPDATE
+                    update!(progress, args...; kw...)
+                end
+            end
+        catch err
+            println()
+            # channel closed should only happen from Base.close(pp), which isn't an error
+            if err != Base.closed_exception()
+                bt = catch_backtrace()
+                showerror(stderr, err, bt)
+                println()
+            end
+        finally
+            close(pp)
+        end
+    end
+    return pp
+end
+
+
+"""
+    FakeChannel()
+
+fake RemoteChannel that doesn't put anything anywhere (for allowing overshoot)
+"""
+struct FakeChannel end
+Base.close(::FakeChannel) = nothing
+Base.isready(::FakeChannel) = false
+Distributed.put!(::FakeChannel, _...) = nothing
+
+struct MultipleChannel
+    channel
+    id
+end
+Distributed.put!(mc::MultipleChannel, x) = put!(mc.channel, (mc.id, x...))
+
+mutable struct MultipleProgress
+    channel
+    amount::Int
+end
+
+Base.getindex(mp::MultipleProgress, n) = ParallelProgress.(MultipleChannel.(Ref(mp.channel), n))
+Base.lastindex(mp::MultipleProgress) = mp.amount
```

Comment:
You should also have `Base.firstindex(mp::MultipleProgress) = 1` and maybe support the iteration protocol?

Reply:
Should it be a subtype of `AbstractVector{ParallelProgress}`?   That way you get iteration etcetera for free (once you implement `size`)

Reply:
I thought about that and wasn't sure how to do it. Right now `mp[0]` is the main progressbar, which means if you want to update all progressbars without the main one, you do `mp[1:end]` and if you want to include the main one you can do `mp[0:end]`.